packages/camera/android/src/main/java/io/flutter/plugins/camera/CameraPlugin.java

```diff
@@ -71,12 +62,23 @@
   // The code to run after requesting camera permissions.
   private Runnable cameraPermissionContinuation;
   private boolean requestingPermission;
+  private final OrientationEventListener orientationEventListener;
+  private int currentOrientation;
 
   private CameraPlugin(Registrar registrar, FlutterView view, Activity activity) {
     this.registrar = registrar;
     this.view = view;
     this.activity = activity;
 
+    orientationEventListener =
+        new OrientationEventListener(activity.getApplicationContext()) {
+          @Override
+          public void onOrientationChanged(int i) {
+            // Convert the raw deg angle to the nearest multiple of 90.
+            currentOrientation = ((i + 45) / 90) * 90;
```

Comment:
Why the 45?

Also, what if `i` is `ORIENTATION_UNKNOWN`?

Reply:
The 45 was because I was doing something overly verbose here. I simplified it out.

This interprets `ORIENTATION_UNKNOWN` as the device pointing straight up. [That's normally sent when the device is basically flat](https://developer.android.com/reference/android/view/OrientationEventListener.html#ORIENTATION_UNKNOWN), so I don't think it's necessarily the wrong thing to just default to that here. Alternatively we could refuse to save orientation with the media at all in this case, or fall back to whatever the screen UI is set to?

Reply:
Decided to not update this value when it's unknown.